pySDC/playgrounds/Diagonal/optimDiagCoeffs.md

```diff
@@ -0,0 +1,57 @@
+|   M | quadType    | distr    | optim   | coeffs                                             |         rho |
+|----:|:------------|:---------|:--------|:---------------------------------------------------|------------:|
+|   2 | GAUSS       | LEGENDRE | QmQd    | (0.105662, 0.394338)                               | 4.1514e-10  |
+|   2 | GAUSS       | LEGENDRE | QmQd    | (0.394338, 0.105662)                               | 3.06546e-10 |
+|   2 | GAUSS       | LEGENDRE | Speck   | (0.166667, 0.5)                                    | 1.44955e-10 |
+|   2 | GAUSS       | LEGENDRE | Speck   | (0.5, 0.166667)                                    | 2.51252e-10 |
+|   2 | LOBATTO     | LEGENDRE | QmQd    | (0.0, 0.5)                                         | 2.47826e-11 |
+|   2 | LOBATTO     | LEGENDRE | Speck   | (0.0, 0.5)                                         | 9.8952e-12  |
+|   2 | RADAU-LEFT  | LEGENDRE | QmQd    | (0.0, 0.333333)                                    | 2.25197e-11 |
+|   2 | RADAU-LEFT  | LEGENDRE | Speck   | (0.0, 0.333333)                                    | 1.11866e-12 |
```

Comment:
Why would the stiff limit and the non-stiff limit yield the same optimisation results? Is it maybe independent of lambda? Maybe it would be interesting to plot the spectral radius of the spectral radius of the iteration matrix with this preconditioner for different values of lambda?

Reply:
That's because in this case, you are doing only one-dimensional optimization. Since the left node coincide with the initial time-step time, then you are looking for a diagonal $Q_{\Delta}$ matrix of the form

```math
Q_{\Delta,D} = \begin{pmatrix} 0 & 0 \\ 0 & \alpha \end{pmatrix}
```

while optimizing the spectral radius of the truncated system $1 - \alpha^-1 Q[1,1]$ (for Speck optimization) and $Q[1,1] - \alpha$ (for QmQd optimization). So in this case, taking $\alpha=Q[1,1]$ does give you the optimized coefficient, see :

```python
>>> genCollocation(2, 'LEGENDRE', 'LOBATTO')[-1]
array([[0. , 0. ],
       [0.5, 0.5]])
>>> genCollocation(2, 'LEGENDRE', 'RADAU-LEFT')[-1]
array([[0.        , 0.        ],
       [0.33333333, 0.33333333]])
```